### ISSUE TITLE
Ping me when there is release 0.1.7

### ISSUE THREAD

**Issue by vsoch** [urlstechie/urlchecker-action#31]
hey @SuperKogito I saw there was lots of good work today! Could you ping me on this issue when the latest two commits are released? I can then test the released version on the repos where they are needed. Then we can chat more about creating a module, if you are still interested.

**Comment by SuperKogito** [comment]
I won't be able to make the release tonight :/ I still have some fixes to do. I will do it tomorrow and will make sure to ping you here. I am still interested in the module. We can discuss it tomorrow ;)

**Comment by SuperKogito** [comment]
@vsoch  v 0.1.7 is out  :smile:

**Comment by SuperKogito** [comment]
- Can you please confirm, that when trying to import the action; you don't get a nicely formatted call example? 
- v0.1.7 is out, I count on your testing feedback to see if the action works as expected.
- We now have online hosted docs hosted on https://urls-checker.readthedocs.io/
- urlstechie has also a website now: https://urlstechie.github.io/
- Concerning the python package how do you want to approach it? 
- Promoting the tools should be one of the next steps, so if you have any nice suggestions please feel free to share them.

**Comment by vsoch** [comment]
Going to give it a test!

**Comment by vsoch** [comment]
okay, works great! The one kind of site that I'm having issues with (white listed for now) is this, for example:

https://pace.gatech.edu/xsedeosg-architect

If you browse to the site, all seems well, but with requests (even with a human User Agent and retry) it will have an SSL error and ultimately fail. I'm not sure if there is a way to catch this, but for now I'm whitelisting.

For the Python package, my thinking is that we should leave the actions repository as it is for now, and create a repository something along the lines of `urlschecker-python`. Then I can take a first shot at creating the structure for the package, reserving the name on pypi, and migrating the various functions into an organized module. Once that is done, we can do a PR to the URLschecker action that will hugely simplify it to essentially just use the library to run the same action. Then most work / updates can be on the python library, and when there are releases there we would just bump the version of the action, and update the version installed. What do you think?

**Comment by SuperKogito** [comment]
So I actually had this same idea in my mind but I am not sure about only one point: 
- If we use the same package in the action (which will simplify things but also limit the flexibility) will we be able to use the python package as a terminal tool? something like [ffmpeg-normalize](https://github.com/slhck/ffmpeg-normalize) ?

**Comment by vsoch** [comment]
Yes that is exactly my thinking - I'd want to be able to run a command locally that mimics the action. Then using the action is just a wrapper for that.

For sharing / getting the word out, I'd say we would want to first develop (and test) this package, update the action, create a list in the docs / README to show where it's used by, and then write up a blog post (or similar) to share on social media, etc. For tools like this, you really get one shot to make an impression, so I think that it's better to wait until the organization has finished these core libraries, we are sure about the UI and interaction, and we've written up really nice docs to go alongside before sharing.

**Comment by SuperKogito** [comment]
It is good that we agree on that. So I think it is on you to lead this since you have more experience and I will assist you. 
I am not familiar with the promoting aspect; all I have done so far is pot some links about my projects on reddit and some other social media but the impact was not as huge as I hoped for. I totally agree that a good package sells itself so I guess we will work on having a nice package first and see how it goes from there.

**Comment by vsoch** [comment]
okay sounds good! I'll get started soon.

**Comment by vsoch** [comment]
Just an update for some testing - I updated the name of the action (for the previously deployed 0.1.7 under the previous repository name) and the update works!

https://github.com/USRSE/usrse.github.io/pull/179/checks?check_run_id=524640888

So my intuition is that we should (if possible) update previous deploys of the action, just to be careful. I'll update buildtest now.

**Comment by SuperKogito** [comment]
That's great news :smile:

**Comment by vsoch** [comment]
I'm going to close this issue - if there were subpoints you want to continue talking about, let's open a new issue and ping me again
